### PR TITLE
Add molecule tests, Jenkinsfile to repository

### PR DESCRIPTION
**Pull request by nre-ableton**
- Add Python requirements files to repository
- Add ansible-lint configuration file
- Add molecule tests to role
- Add Jenkinsfile to repository
- Build pushes and pull requests on main branch

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -0,0 +1,181 @@
+#
+# This file is autogenerated by pip-compile with python 3.9
```
Comment (ala-ableton) [inline requirements-dev.txt:2]: Please use Python 3.8, like on CI.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        // If only warnings are found, ansible-lint will exit with code 0 but still write
+        // an error summary to stdout. It's not possible to treat warnings as errors, and
+        // likely will never be. See:
+        // https://github.com/ansible-community/ansible-lint/issues/236
+        if (stdout) {
+          error 'ansible-lint exited with warnings, check the output of the previous step'
+        }
+      },
+      black: { data.venv.run('black --check .') },
+      groovylint: { groovylint.check('./Jenkinsfile') },
```
Comment (ala-ableton) [inline Jenkinsfile:30]:
```suggestion
      groovylint: { groovylint.checkSingleFile(path: './Jenkinsfile') },
```

**Review by ala-ableton — CHANGES_REQUESTED** [review]

**Review by ala-ableton — APPROVED** [review]
Once rebased, LGTM 👍🏻

**Comment by nre-ableton** [comment]
@ablbot autosquash

**Pull request merged**